### PR TITLE
Update python-package.yml

### PR DESCRIPTION
**Pull request by andreasprlic**
I think we need this change to fix the deploy action

### CODE REVIEW

**Comment by jsstevenson** [comment]
@korikuzma maybe this is a good time to add the trusted publishing component from the example repo?

**Comment by korikuzma** [comment]
@jsstevenson I agree. @reece would you be able to add trusted publishing on pypi?

**Comment by korikuzma** [comment]
@andreasprlic trusted publishing has been setup on pypi. Would you be able to update the file using the example [here](https://github.com/biocommons/biocommons.example/blob/main/.github/workflows/python-package.yml)? You'll only need to change this line: https://github.com/biocommons/biocommons.example/blob/main/.github/workflows/python-package.yml#L129

**Comment by bsgiles73** [comment]
@reece I was not able to reproduce the local unit test error that you ran into in today's collective coding meeting.
```
(3.11) sgiles-MD6M:bioutils shane.giles$ make test
Using Python 3.11
pytest
============================================================================================= test session starts =============================================================================================
platform darwin -- Python 3.11.4, pytest-7.4.4, pluggy-1.5.0 -- /Users/shane.giles/locus/opt/repos/biocommons/bioutils/venv/3.11/bin/python3.11
cachedir: .pytest_cache
rootdir: /Users/shane.giles/locus/opt/repos/biocommons/bioutils
configfile: pyproject.toml
testpaths: tests
plugins: optional-tests-0.1.1, cov-4.1.0
collected 22 items

tests/test_normalize.py::test_trim[normalize_fn0] PASSED
tests/test_normalize.py::test_trim[normalize_fn1] PASSED
tests/test_normalize.py::test_anchor[normalize_fn0] PASSED
tests/test_normalize.py::test_anchor[normalize_fn1] PASSED
tests/test_normalize.py::test_trinuc PASSED
tests/test_normalize.py::test_bounds PASSED
tests/test_normalize.py::test_no_trim_no_shuffle PASSED
tests/test_normalize.py::test_shuffle_no_trim PASSED
tests/test_normalize.py::test_mode_string PASSED
tests/test_normalize.py::test_input_alleles_not_modified PASSED
tests/test_normalize.py::test_error_distinct[normalize_fn0] PASSED
tests/test_normalize.py::test_error_distinct[normalize_fn1] PASSED
tests/test_normalize.py::test_error_ref_allele PASSED
tests/test_normalize.py::test_error_vcf_mode_no_trim PASSED
tests/test_seqfetcher.py::test_fetch_seq PASSED
tests/test_seqfetcher.py::test_fetch_seq_ncbi_invalid_positions PASSED
tests/test_seqfetcher.py::test_add_eutils_api_key PASSED
tests/test_seqfetcher.py::test_fetch_seq_errors PASSED
tests/test_seqfetcher.py::test_rate_limit PASSED
tests/test_sequences.py::test_translate_examples PASSED
tests/test_sequences.py::test_translate_selenoproteins PASSED
tests/test_sequences.py::test_translate_vertebrate_mitochondrial PASSED/Users/shane.giles/locus/opt/repos/biocommons/bioutils/venv/3.11/lib/python3.11/site-packages/coverage/control.py:894: CoverageWarning: No data was collected. (no-data-collected)
  self._warn("No data was collected.", slug="no-data-collected")


---------- coverage: platform darwin, python 3.11.4-final-0 ----------
Name                              Stmts   Miss  Cover
-----------------------------------------------------
src/bioutils/__init__.py              3      3     0%
src/bioutils/_versionwarning.py       7      7     0%
src/bioutils/accessions.py           36     36     0%
src/bioutils/assemblies.py           21     21     0%
src/bioutils/coordinates.py           7      7     0%
src/bioutils/cytobands.py            14     14     0%
src/bioutils/digest.py               25     25     0%
src/bioutils/digests.py              28     28     0%
src/bioutils/exceptions.py            2      2     0%
src/bioutils/normalize.py           131    131     0%
src/bioutils/seqfetcher.py           71     71     0%
src/bioutils/sequences.py           101    101     0%
src/bioutils/vmc_digest.py           11     11     0%
-----------------------------------------------------
TOTAL                               457    457     0%


============================================================================================= 22 passed in 1.94s ==============================================================================================
```

**Comment by korikuzma** [comment]
@bsgiles73 same

**Comment by andreasprlic** [comment]
@korikuzma I replaced the python-package.yml with the one from the template and updated the pypi bioutils URL accordingly. It looks like it needs a few more tweaks though.

**Review by korikuzma — APPROVED** [review]

**Review by korikuzma — CHANGES_REQUESTED** [review]
Oh my bad, didn't see tests are failing now

**Comment by andreasprlic** [comment]
@korikuzma  after a few more tweaks (in Makefile), tests are passing.

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -1,47 +1,48 @@
...
+        python-version: "3.12"
         cache: pip
-        cache-dependency-path: '**/setup.cfg'
+        cache-dependency-path: '**/pyproject.yaml'
 
     - name: Install test dependencies
       run: |
         python -m pip install --upgrade pip
-        pip install --use-deprecated=legacy-resolver -e .[dev]
+        pip install --use-deprecated=legacy-resolver -e .[dev,test]
```
Comment (korikuzma) [inline .github/workflows/python-package.yml:37]: It might be good to consider breaking out test vs dev deps in the example repo

**Review by korikuzma — APPROVED** [review]

File: .github/workflows/python-package.yml (hunk 1 above)
Comment (jsstevenson) [reply .github/workflows/python-package.yml:37]: draft PR: https://github.com/biocommons/biocommons.example/pull/53

**Pull request merged**